packages/salesforce-adapter/src/filters/flows_filter.ts

```diff
@@ -47,62 +47,103 @@ import { SalesforceRecord } from '../client/types'
 const { isDefined } = lowerdashValues
 
 const log = logger(module)
+const { DefaultMap } = collections.map
 const { toArrayAsync } = collections.asynciterable
 
 const DEFAULT_CHUNK_SIZE = 500
 
-const fixFilePropertiesName = (props: FileProperties, activeVersions: Map<string, string>): FileProperties => ({
-  ...props,
-  fullName: activeVersions.get(`${props.fullName}`) ?? `${props.fullName}`,
-})
+type FlowVersionRecord = SalesforceRecord & {
+  DefinitionId: string
+  VersionNumber: number
+  Status: string
+  CreatedDate: string
+  CreatedBy: {
+    Name: string
+  }
+  LastModifiedDate: string
+  LastModifiedBy: {
+    Name: string
+  }
+}
+const isFlowVersionRecord = (record: SalesforceRecord): record is FlowVersionRecord =>
+  _.isString(record.DefinitionId) &&
+  _.isString(record.Status) &&
+  _.isNumber(record.VersionNumber) &&
+  _.isString(record.CreatedDate) &&
+  _.isPlainObject(record.CreatedBy) &&
+  _.isString(record.CreatedBy.Name) &&
+  _.isString(record.LastModifiedDate) &&
+  _.isPlainObject(record.LastModifiedBy) &&
+  _.isString(record.LastModifiedBy.Name)
 
-type FlowDefinitionViewRecord = SalesforceRecord & {
-  ActiveVersionId: string | null
-  ApiName: string
+type FlowVersionProperties = {
+  id: string
+  version: number
+  status: string
+  createdByName: string
+  createdDate: string
+  lastModifiedByName: string
+  lastModifiedDate: string
 }
-const isFlowDefinitionViewRecord = (record: SalesforceRecord): record is FlowDefinitionViewRecord =>
-  (record.ActiveVersionId === null || _.isString(record.ActiveVersionId)) && _.isString(record.ApiName)
 
-const getActiveFlowVersionIdByApiName = async ({
+const getFlowVersionsByApiName = async ({
   client,
   flowDefinitions,
   chunkSize,
 }: {
   client: SalesforceClient
   flowDefinitions: InstanceElement[]
   chunkSize: number
-}): Promise<Record<string, string>> => {
-  const flowDefinitionsIds = flowDefinitions.map(flow => flow.value[INTERNAL_ID_FIELD])
+}): Promise<Map<string, FlowVersionProperties[]>> => {
+  const flowDefinitionsIdsToApiNames = Object.fromEntries(
```

Comment:
```suggestion
  const flowDefinitionsIdsToApiNames = Object.fromEntries<string>(
```
Specify the value type explicitly, or add typeguard (might be overkill as instances should always have fullName string field) otherwise the tpye of **flowDefinitionsIdsToApiNames** will be `any`

Reply:
Nice catch